homeassistant/components/websocket_api/commands.py

```diff
@@ -231,19 +235,68 @@ async def handle_call_service(
         connection.send_result(msg["id"], {"context": context})
     except ServiceNotFound as err:
         if err.domain == msg["domain"] and err.service == msg["service"]:
-            connection.send_error(msg["id"], const.ERR_NOT_FOUND, "Service not found.")
+            err.translation_key = "service_not_found_general"
```

Comment:
I was suprised when I saw this, that we set the translation key in the place where we handle the exception. If that is needed I'm wondering if the exception is the correct place to hold the translation info.

Reply:
Alternative this could be set in the class definition. But it seems we have 2 variants with different descriptions.

Reply:
This seems the only places where we raise a `ServiceNotFound` error.

https://github.com/home-assistant/core/blob/1a1bc05470c431094cb6f48c38c95aa7444de2a0/homeassistant/core.py#L1980